relay-spans/src/span.rs

```diff
@@ -0,0 +1,350 @@
+use std::str::FromStr;
+
+use chrono::{TimeZone, Utc};
+use serde::Deserialize;
+use serde_repr::Deserialize_repr;
+
+use relay_event_schema::protocol::{Span as EventSpan, SpanId, SpanStatus, Timestamp, TraceId};
+use relay_protocol::{Annotated, Object, Value};
+
+use crate::status_codes;
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Span {
+    pub trace_id: String,
+    pub span_id: String,
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub trace_state: String,
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub parent_span_id: Option<String>,
+    pub name: String,
+    pub kind: SpanKind,
+    pub start_time_unix_nano: i64,
+    pub end_time_unix_nano: i64,
```

Comment:
What formal schema are we implementing? Is it this one? https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto.  cc @mydea 

Asking because on [this page](https://opentelemetry.io/docs/concepts/signals/traces/#spans) I only see `start_time` and `end_time` as strings, no `_unix_nano` suffix.

Reply:
It's indeed this one we're implementing. All SDKs auto-generated those structures and would send us a JSON structure based on this protobuf file.

The other page you link is the more user friendly implementation to build a span in the SDK.

Ideally, I would want us to use the auto-generated version eventually but they seemed to have only files meant to be used with their GPRC/tonic implementations and I wanted serde + JSON. Eventually it'd be good to migrate to something auto-generated from the protobuf though.

Reply:
hmm good point, I generated the example based on what opentelemetry JS exports for OTLP 🤔